cms/search/signal_handlers.py

```diff
@@ -57,12 +57,16 @@ def on_page_deleted(sender: "type[Page]", instance: "Page", **kwargs: dict) -> N
 def on_page_moved(sender: "type[Page]", instance: "Page", **kwargs: dict) -> None:  # pylint: disable=unused-argument
     """Called whenever a Wagtail Page is moved in the tree (UI or code).
     instance is the moved Page object.
-    We use the publish_created_or_updated method to update search for the moved page and any of it's non-excluded
+    We use the publish_created_or_updated method to update search for the moved page and any of its non-excluded
     descendants, which will also be affected by the move.
     """
     if kwargs["url_path_before"] == kwargs["url_path_after"]:
         # No change in URL path, no need to update search index of the instance or descendants
         return
+    if instance.get_view_restrictions().exists():
+        # Pages with view restrictions should not be exposed in search
+        # this is inherited by descendants, so nothing more to do
+        return
```

Comment:
Nit, let's add a new property to the base page `live_and_public`?. Is the instance `Page` or our `BasePage`? If it is not our page, it can't be a property.? 

The `live` check should be done first to avoid querying the DB to check for restrictions on draft pages.

Reply:
There are some edge cases, i.e. what if a parent page restriction was changed from private to public, post move. Only the parent page would be sent for search indexing.

Reply:
I think the page published receiver needs to handle that edge case, there are other similar edge cases around changing a page with descendants, such as editing the slug.

Reply:
The on publish receiver doesn't currently consider descendants, maybe another card needed.